### PR TITLE
fix: Using Redis for rate limiting

### PR DESCRIPTION
**Pull request by Rajakavitha1**
### Description

Formatting changes and curl option changes.
Used placeholders because the command varies based on the install method

### Testing instructions

Preview link: 

### Checklist 

- [ ] Review label added 
- [ ] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

### CODE REVIEW

File: app/_src/kubernetes-ingress-controller/guides/redis-rate-limiting.md [hunk 1]
```diff
@@ -28,183 +20,162 @@ and use it for your plugin instead of the example rate-limiting plugin.
...
-]'
-```
-
-Replace `PASSWORD` with your actual password.
+for Redis with turnkey options for authentication.
+
+1.  Create a password Secret and replace `<PASSWORD>` with a password of your choice.
+
+    ```bash
+    kubectl create -n kong secret generic redis-password-secret --from-literal=redis-password=<PASSWORD>
```
Comment (mheap) [inline app/_src/kubernetes-ingress-controller/guides/redis-rate-limiting.md:134]: I get an error due to the `<>`.

```
kubectl create -n kong secret generic redis-password-secret --from-literal=redis-password=<PASSWORD>

zsh: parse error near `\n'
```

File: app/_src/kubernetes-ingress-controller/guides/redis-rate-limiting.md [hunk 2]
```diff
@@ -279,7 +249,7 @@ kubectl patch kongplugin rate-limit --type json --patch '[
 
 ### Check plugin configuration
 
-The updated KongPlugin will propagate the reference to the proxy configuration.
+The updated KongPlugin propagates the reference to the proxy configuration.
 You can confirm it by checking the admin API.
 
 In one terminal, open a port-forward to the admin API:
```
Comment (mheap) [inline app/_src/kubernetes-ingress-controller/guides/redis-rate-limiting.md:255]: These instructions don't work. The deployment is called `kong-gateway` not `ingress-kong`

It's also missing a `-n kong`

File: app/_src/kubernetes-ingress-controller/guides/redis-rate-limiting.md [hunk 3]
```diff
@@ -329,3 +298,4 @@ The results should look like this:
 < X-RateLimit-Remaining-Minute: 0
 < X-RateLimit-Remaining-Minute: 0
 ```
+The decrement counters sequentially regardless of the {{site.base_gateway}} replica count.
```
Comment (mheap) [inline app/_src/kubernetes-ingress-controller/guides/redis-rate-limiting.md:301]:
```suggestion
The counters decrement sequentially regardless of the {{site.base_gateway}} replica count.
```

File: app/_src/kubernetes-ingress-controller/guides/redis-rate-limiting.md (hunk 2 above)
Comment (mheap) [reply app/_src/kubernetes-ingress-controller/guides/redis-rate-limiting.md:255]: I can't add a comment to the line as it hasn't changed.

```
"{vault://env/secret-redis-password}"
Forwarding from 127.0.0.1:8444 -> 8444
```

Should be:

```
Forwarding from 127.0.0.1:8444 -> 8444
```

File: app/_includes/md/kic/prerequisites.md [hunk 4]
```diff
@@ -36,9 +36,9 @@ Kubernetes exposes the proxy through a Kubernetes service. Run the following com
 1. Populate `$PROXY_IP` for future commands:
 
     ```bash
-    export PROXY_IP=$(kubectl get -o jsonpath="{.status.loadBalancer.ingress[0].ip}" service -n kong kong-gateway-proxy)
-    
-    echo "Proxy IP: $PROXY_IP"
```
Comment (mheap) [inline app/_includes/md/kic/prerequisites.md:41]: I found the `echo` here useful

File: app/_includes/md/kic/prerequisites.md [hunk 5]
```diff
@@ -36,9 +36,9 @@ Kubernetes exposes the proxy through a Kubernetes service. Run the following com
...
+    HOST=$(kubectl get svc --namespace kong kong-gateway-proxy -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+    PORT=$(kubectl get svc --namespace kong kong-gateway-proxy -o jsonpath='{.spec.ports[0].port}')
```
Comment (mheap) [inline app/_includes/md/kic/prerequisites.md:40]: Why do we add the port? Don't loadbalancers always listen on port 80 by default?

**Review by mheap — CHANGES_REQUESTED** [review]
Most of the guide works. I left inline comments where applicable.

I'm not clear on why we set up the Redis password as a string, then update it to be a vault reference. This feels like two guides to me. One for configuring and using Redis, and another on configuring plugins using Secrets Management

File: app/_includes/md/kic/prerequisites.md (hunk 5 above)
Comment (Rajakavitha1) [reply app/_includes/md/kic/prerequisites.md:40]: I just added the commands that I saw after installing using Helm.

File: app/_src/kubernetes-ingress-controller/guides/redis-rate-limiting.md [hunk 6]
```diff
@@ -28,183 +20,162 @@ and use it for your plugin instead of the example rate-limiting plugin.
...
-    "path":"/config/redis_password",
-    "value":"PASSWORD"
-  }
-]'
-```
-
-Replace `PASSWORD` with your actual password.
+for Redis with turnkey options for authentication.
+
+1.  Create a password Secret and replace `<PASSWORD>` with a password of your choice.
```
Comment (mheap) [inline app/_src/kubernetes-ingress-controller/guides/redis-rate-limiting.md:131]:
```suggestion
1.  Create a password Secret and replace `PASSWORD` with a password of your choice.
```

File: app/_src/kubernetes-ingress-controller/guides/redis-rate-limiting.md [hunk 7]
```diff
@@ -28,183 +20,162 @@ and use it for your plugin instead of the example rate-limiting plugin.
...
+    ```bash
+    kubectl create -n kong secret generic redis-password-secret --from-literal=redis-password=PASSWORD
+    ```
+    The results should look like this:
+
+    ```text
+    secret/redis-password-secret created
+    ```
+
+1. Install a chart release using the Secret.
```
Comment (mheap) [inline app/_src/kubernetes-ingress-controller/guides/redis-rate-limiting.md:142]:
```suggestion
1. Install Redis
```

File: app/_src/kubernetes-ingress-controller/guides/redis-rate-limiting.md [hunk 8]
```diff
@@ -28,183 +20,162 @@ and use it for your plugin instead of the example rate-limiting plugin.
...
+
+    ```bash
+    helm install -n kong redis oci://registry-1.docker.io/bitnamicharts/redis \
+      --set auth.existingSecret=redis-password-secret \
+      --set architecture=standalone
+    ```
+    Helm displays the instructions that describes the new installation.
+
+1. Update your plugin configuration with the `redis` policy, Service, and credentials. Replace `<PASSWORD>` with the password that you set for Redis.
```
Comment (mheap) [inline app/_src/kubernetes-ingress-controller/guides/redis-rate-limiting.md:151]:
```suggestion
1. Update your plugin configuration with the `redis` policy, Service, and credentials. Replace `PASSWORD` with the password that you set for Redis.
```

File: app/_src/kubernetes-ingress-controller/guides/redis-rate-limiting.md [hunk 9]
```diff
@@ -307,13 +276,12 @@ The results should look like this:
 ```
 {% endif_version %}
 
-## Test it
+## Test rate limiting is a multi-node Kong deployment
```
Comment (mheap) [inline app/_src/kubernetes-ingress-controller/guides/redis-rate-limiting.md:279]: I'd move this section above the optional Secrets Management section

**Review by mheap — CHANGES_REQUESTED** [review]

**Comment by Rajakavitha1** [comment]
> Most of the guide works. I left inline comments where applicable.
> 
> I'm not clear on why we set up the Redis password as a string, then update it to be a vault reference. This feels like two guides to me. One for configuring and using Redis, and another on configuring plugins using Secrets Management

Yes!!! That is right. just so that you get some context. Here's the PR on which this guide was written: https://github.com/Kong/docs.konghq.com/pull/6069

File: app/_src/kubernetes-ingress-controller/guides/redis-rate-limiting.md (hunk 9 above)
Comment (Rajakavitha1) [reply app/_src/kubernetes-ingress-controller/guides/redis-rate-limiting.md]: Done!

File: app/_src/kubernetes-ingress-controller/guides/redis-rate-limiting.md [hunk 10]
```diff
@@ -28,183 +20,185 @@ and use it for your plugin instead of the example rate-limiting plugin.
...
+    ```bash
+    kubectl scale --replicas 3 -n kong deployment kong-gateway
+    ```
+    The results should look like this:
+    ```text
+    deployment.apps/kong-gateway scaled
+    ```
+1.  Check if the status of all the Pods that are `READY` is `Running` using the command `kubectl get pods -n kong`.
+
+1. Send requests to this Service does not reliably decrement the remaining counter.
```
Comment (mheap) [inline app/_src/kubernetes-ingress-controller/guides/redis-rate-limiting.md:99]:
```suggestion
1. Sending requests to this Service does not reliably decrement the remaining counter.
```

File: app/_src/kubernetes-ingress-controller/guides/redis-rate-limiting.md [hunk 11]
```diff
@@ -28,183 +20,185 @@ and use it for your plugin instead of the example rate-limiting plugin.
...
+      },
+      {
+        "op":"add",
+        "path":"/config/redis_host",
+        "value":"redis-master"
+      },
+      {
+        "op":"add",
+        "path":"/config/redis_password",
+        "value":"<PASSWORD>"
```
Comment (mheap) [inline app/_src/kubernetes-ingress-controller/guides/redis-rate-limiting.md:168]:
```suggestion
        "value":"PASSWORD"
```

**Review by mheap — CHANGES_REQUESTED** [review]

**Review by mheap — APPROVED** [review]

**Pull request merged**